socorro/external/es/connection_context.py

```diff
@@ -93,24 +98,49 @@ def __init__(self, config):
     def connection(self, name=None, timeout=None):
         """Returns an instance of elasticsearch-py's Elasticsearch class as
         encapsulated by the Connection class above.
+
         Documentation: http://elasticsearch-py.readthedocs.org
+
         """
         if timeout is None:
             timeout = self.config.elasticsearch_timeout
 
+        # Separate the HTTP from HTTPS urls so we know if they're all one or
+        # the other because we don't want a mix
+        http = [url for url in self.config.elasticsearch_urls if not url.startswith('https')]
+        https = [url for url in self.config.elasticsearch_urls if url.startswith('https')]
```

Comment:
The `httpspf700://` protocol passes this test. I'm not sure if that matters, but it smells a bit like the kind of things that combine to form a security vulnerability.

Also, the check above includes non-http protocols in `http`.

We should probably be parsing out the protocol here instead of relying on `startswith`.

Reply:
A while back, you could also use thrift, but I'm pretty sure Elasticsearch doesn't support that anymore. I think the elasticsearch-py library only supports http/https through requests and urllib3.

However, I'll switch it to parse the url instead--that's a good idea.